File: unity/Assets/Scripts/Multiplayer/Time/ServerTimeResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UnityEngine;

namespace Michitai.Multiplayer.Time
{
    /// <summary>
    /// Response containing the current server time in UTC.
    /// </summary>
    [System.Serializable]
    public class ServerTimeResponse : ApiResponse<ETimeError>
    {
        [SerializeField]
        private string utc;

        /// <summary>
        /// Unix timestamp of the current server time.
        /// </summary>
        public long timestamp;

        /// <summary>
        /// Human-readable format of the server time.
        /// </summary>
        public string readable;

        /// <summary>
        /// The current server time in UTC (parsed from string).
        /// </summary>
        public DateTimeOffset? Utc
        {
            get
            {
                return Time.ParseUtc(utc);
            }
        }
    }
}

File: java/src/main/java/com/michitai/multiplayer/time/ServerTimeResponse.java
package com.michitai.multiplayer.time;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

/**
 * Response containing the server UTC time, timestamp, and readable format.
 */
public class ServerTimeResponse extends ApiResponse {
    @JsonProperty("utc")
    private String utc;

    @JsonProperty("timestamp")
    private long timestamp;

    @JsonProperty("readable")
    private String readable;

    public String getUtc() {
        return utc;
    }

    public void setUtc(String utc) {
        this.utc = utc;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }

    public String getReadable() {
        return readable;
    }

    public void setReadable(String readable) {
        this.readable = readable;
    }
}
